File: backend/src/Shalom.Application/Contracts/FastingScheduleDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// The schedule-first fasting configuration (ADR-005) plus the computed
/// <paramref name="TodayWindow"/> for the server-derived local day (ADR-004).
/// </summary>
public record FastingScheduleDto(
    TimeOnly EatingWindowStart,
    TimeOnly EatingWindowEnd,
    int TargetFastHours,
    string TimeZoneId,
    IReadOnlyList<FastingOverrideDto> Overrides,
    TodayWindowDto TodayWindow);

File: frontend/projects/api/src/lib/models/fasting-schedule-dto.ts
import { FastingOverrideDto } from './fasting-override-dto';
import { TodayWindowDto } from './today-window-dto';

/**
 * The schedule-first fasting configuration (ADR-005) plus the computed
 * window for the server-derived local day. Times are `HH:mm:ss`.
 */
export interface FastingScheduleDto {
  readonly eatingWindowStart: string;
  readonly eatingWindowEnd: string;
  readonly targetFastHours: number;
  /** IANA id, e.g. "America/Toronto". */
  readonly timeZoneId: string;
  readonly overrides: FastingOverrideDto[];
  readonly todayWindow: TodayWindowDto;
}
